injected/src/features/message-bridge.js

```diff
@@ -0,0 +1,168 @@
+    }
+
+    /**
+     * @param {ProxyRequest} request
+     * @param {(s: string) => string} appendToken
+     */
+    proxyRequest(request, appendToken) {
+        const { id, featureName, method, params } = request;
+
+        const proxy = this.proxies.get(featureName);
+        if (!proxy) return console.warn('[isolated] proxy was not installed');
+
+        const responseName = appendToken(ProxyResponse.NAME + '-' + id);
+        console.log('[isolated] will proxy', request);
+
+        proxy
+            ?.request(method, params)
+            // eslint-disable-next-line promise/prefer-await-to-then
+            .then((response) => {
+                const responseEvent = new ProxyResponse({
+                    method,
+                    featureName,
+                    result: response,
+                    id,
+                });
+                const evt = new CustomEvent(responseName, { detail: responseEvent });
+                console.log('[isolated] will dispatch', evt);
+                window.dispatchEvent(evt);
+            })
+            // eslint-disable-next-line promise/prefer-await-to-then
+            .catch((e) => {
+                const errorResponseEvent = new ProxyResponse({
+                    method,
+                    featureName,
+                    error: { message: e.message },
+                    id,
+                });
+                const evt = new CustomEvent(responseName, { detail: errorResponseEvent });
+                window.dispatchEvent(evt);
+            });
+    }
+
+    /**
+     * @param {SubscriptionRequest} subscription
+     * @param {(s: string) => string} appendToken
+     */
+    proxySubscription(subscription, appendToken) {
+        const { id, featureName, subscriptionName } = subscription;
+        const proxy = this.proxies.get(subscription.featureName);
+        if (!proxy) return console.warn('[isolated] proxy was not installed');
+
+        const responseName = appendToken(SubscriptionResponse.NAME + '-' + id);
+
+        console.log('[isolated] will setup subscription', subscription);
+        this.subscriptions.set(
+            id,
+            proxy.subscribe(subscriptionName, (data) => {
+                const responseEvent = new SubscriptionResponse({
+                    subscriptionName,
+                    featureName,
+                    params: data,
+                    id,
+                });
+                const evt = new CustomEvent(responseName, { detail: responseEvent });
+                console.log('[isolated] will dispatch SubscriptionResponse', evt);
+                window.dispatchEvent(evt);
+            }),
+        );
+    }
+
+    /**
+     * @param {SubscriptionUnsubscribe} unsubscribe
+     */
+    removeSubscription(unsubscribe) {
+        const { id } = unsubscribe;
+        const fn = this.subscriptions.get(id);
+        console.log(`[isolated] will remove subscription`, id);
+        fn?.();
```

Comment:
Is this indended?

Reply:
I think we need this to actually unsubscribe in the messaging code, don't we?

Reply:
yep exactly, I renamed it to make it clearer.